server: Provide a more useful error if init is run more than once

a-robinson opened the pull request:
It now prints confirmation of success when it succeeds and a more
digestible error when run on an already initialized cluster.

Also, fix the acceptance test framework as necessary to make the test function.

Fixes #18484.

a-robinson committed:
acceptance: Fix DockerCluster.ExecCLI to match the expected behavior

The Cluster interface requires that the ExecCLI implementation prepends
the cockroach binary before the args passed to the command.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18489)
<!-- Reviewable:end -->

bdarnell commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/cli/init.go, line 57 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18489#-Ku-rlS4GVeq_Y8goeXx:-Ku-rlS4GVeq_Y8goeXy:b4bnw3w) ([raw file](https://github.com/cockroachdb/cockroach/blob/d456c655589676c7b606e589f9b49e3528cfb7fe/pkg/cli/init.go#L57)):*
> ```Go
> 	if _, err = c.Bootstrap(ctx, &serverpb.BootstrapRequest{}); err != nil {
> 		// We'll only get an unimplemented error if we successfully connected to the
> 		// server but the server has already closed its "Init" service. In such
> ```

I think I'd rather provide an "Init" service for already-initialized clusters that just returns this error. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18489)*
<!-- Sent from Reviewable.io -->

a-robinson committed:
server: Provide a more useful error if init is run more than once

`cockroach init` now prints confirmation of success when it succeeds
and a more digestible error when run on an already initialized cluster.

a-robinson commented:
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/cli/init.go, line 57 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18489#-Ku-rlS4GVeq_Y8goeXx:-Ku5kHCHcksEnzj50dJA:b-86etsm) ([raw file](https://github.com/cockroachdb/cockroach/blob/d456c655589676c7b606e589f9b49e3528cfb7fe/pkg/cli/init.go#L57)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

I think I'd rather provide an "Init" service for already-initialized clusters that just returns this error. 
</blockquote></details>

Yeah, that is a better approach. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18489)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1, 1 of 2 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18489#-:-KuKp7OxudB9Satlrs6y:bnfp4nl)*
<!-- Sent from Reviewable.io -->
